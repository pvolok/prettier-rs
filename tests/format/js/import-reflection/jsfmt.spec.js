run_spec(import.meta, ["babel", "flow", "typescript"], {
  errors: {
    flow: [
      "comments.js",
      "import-reflection.js",
      "valid-default-import.mjs",
      "valid-from-as-default-module-binding-escaped.mjs",
      "valid-from-as-default-module-binding.mjs",
    ],
    typescript: [
      "comments.js",
      "import-reflection.js",
      "valid-default-import.mjs",
      "valid-from-as-default-module-binding-escaped.mjs",
      "valid-from-as-default-module-binding.mjs",
    ],
    acorn: [
      "comments.js",
      "import-reflection.js",
      "valid-default-import.mjs",
      "valid-from-as-default-module-binding-escaped.mjs",
      "valid-from-as-default-module-binding.mjs",
    ],
    espree: [
      "comments.js",
      "import-reflection.js",
      "valid-default-import.mjs",
      "valid-from-as-default-module-binding-escaped.mjs",
      "valid-from-as-default-module-binding.mjs",
    ],
    meriyah: [
      "comments.js",
      "import-reflection.js",
      "valid-default-import.mjs",
      "valid-from-as-default-module-binding-escaped.mjs",
      "valid-from-as-default-module-binding.mjs",
    ],
  },
});
