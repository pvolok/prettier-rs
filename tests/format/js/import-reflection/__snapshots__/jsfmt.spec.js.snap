// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comments.js [acorn] format 1`] = `
"Unexpected token (1:48)
> 1 | /* 0 */import /* 1 */module /* 2 */from /* 3 */from /* 4 */"./module.wasm"/* 5 */;
    |                                                ^
  2 |"
`;

exports[`comments.js [espree] format 1`] = `
"Unexpected token from (1:48)
> 1 | /* 0 */import /* 1 */module /* 2 */from /* 3 */from /* 4 */"./module.wasm"/* 5 */;
    |                                                ^
  2 |"
`;

exports[`comments.js [flow] format 1`] = `
"Unexpected identifier, expected a string (1:48)
> 1 | /* 0 */import /* 1 */module /* 2 */from /* 3 */from /* 4 */"./module.wasm"/* 5 */;
    |                                                ^^^^
  2 |"
`;

exports[`comments.js [meriyah] format 1`] = `
"Import source must be string (1:51)
> 1 | /* 0 */import /* 1 */module /* 2 */from /* 3 */from /* 4 */"./module.wasm"/* 5 */;
    |                                                   ^
  2 |"
`;

exports[`comments.js [typescript] format 1`] = `
"';' expected. (1:60)
> 1 | /* 0 */import /* 1 */module /* 2 */from /* 3 */from /* 4 */"./module.wasm"/* 5 */;
    |                                                            ^
  2 |"
`;

exports[`comments.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
/* 0 */import /* 1 */module /* 2 */from /* 3 */from /* 4 */"./module.wasm"/* 5 */;

=====================================output=====================================
/* 0 */ import module /* 1 */ /* 2 */ from /* 3 */ from /* 4 */ "./module.wasm" /* 5 */;

================================================================================
`;

exports[`import-reflection.js [acorn] format 1`] = `
"Unexpected token (1:15)
> 1 | import module foo from "./module.wasm";
    |               ^
  2 |"
`;

exports[`import-reflection.js [espree] format 1`] = `
"Unexpected token foo (1:15)
> 1 | import module foo from "./module.wasm";
    |               ^
  2 |"
`;

exports[`import-reflection.js [flow] format 1`] = `
"Unexpected identifier, expected the keyword \`from\` (1:15)
> 1 | import module foo from "./module.wasm";
    |               ^^^
  2 |"
`;

exports[`import-reflection.js [meriyah] format 1`] = `
"Import source must be string (1:17)
> 1 | import module foo from "./module.wasm";
    |                 ^
  2 |"
`;

exports[`import-reflection.js [typescript] format 1`] = `
"'=' expected. (1:15)
> 1 | import module foo from "./module.wasm";
    |               ^
  2 |"
`;

exports[`import-reflection.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
import module foo from "./module.wasm";

=====================================output=====================================
import module foo from "./module.wasm";

================================================================================
`;

exports[`valid-default-import.mjs [acorn] format 1`] = `
"Unexpected token (1:15)
> 1 | import module foo from "./module.wasm";
    |               ^
  2 | import bar from "./module.wasm";
  3 |"
`;

exports[`valid-default-import.mjs [espree] format 1`] = `
"Unexpected token foo (1:15)
> 1 | import module foo from "./module.wasm";
    |               ^
  2 | import bar from "./module.wasm";
  3 |"
`;

exports[`valid-default-import.mjs [flow] format 1`] = `
"Unexpected identifier, expected the keyword \`from\` (1:15)
> 1 | import module foo from "./module.wasm";
    |               ^^^
  2 | import bar from "./module.wasm";
  3 |"
`;

exports[`valid-default-import.mjs [meriyah] format 1`] = `
"Import source must be string (1:17)
> 1 | import module foo from "./module.wasm";
    |                 ^
  2 | import bar from "./module.wasm";
  3 |"
`;

exports[`valid-default-import.mjs [typescript] format 1`] = `
"'=' expected. (1:15)
> 1 | import module foo from "./module.wasm";
    |               ^
  2 | import bar from "./module.wasm";
  3 |"
`;

exports[`valid-default-import.mjs format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
import module foo from "./module.wasm";
import bar from "./module.wasm";

=====================================output=====================================
import module foo from "./module.wasm";
import bar from "./module.wasm";

================================================================================
`;

exports[`valid-from-as-default-module-binding.mjs [acorn] format 1`] = `
"Unexpected token (1:20)
> 1 | import module from from "./module.wasm";
    |                    ^
  2 |"
`;

exports[`valid-from-as-default-module-binding.mjs [espree] format 1`] = `
"Unexpected token from (1:20)
> 1 | import module from from "./module.wasm";
    |                    ^
  2 |"
`;

exports[`valid-from-as-default-module-binding.mjs [flow] format 1`] = `
"Unexpected identifier, expected a string (1:20)
> 1 | import module from from "./module.wasm";
    |                    ^^^^
  2 |"
`;

exports[`valid-from-as-default-module-binding.mjs [meriyah] format 1`] = `
"Import source must be string (1:23)
> 1 | import module from from "./module.wasm";
    |                       ^
  2 |"
`;

exports[`valid-from-as-default-module-binding.mjs [typescript] format 1`] = `
"';' expected. (1:25)
> 1 | import module from from "./module.wasm";
    |                         ^
  2 |"
`;

exports[`valid-from-as-default-module-binding.mjs format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
import module from from "./module.wasm";

=====================================output=====================================
import module from from "./module.wasm";

================================================================================
`;

exports[`valid-from-as-default-module-binding-escaped.mjs [acorn] format 1`] = `
"Unexpected token (1:15)
> 1 | import module \\u0066rom from "./module.wasm";
    |               ^
  2 |"
`;

exports[`valid-from-as-default-module-binding-escaped.mjs [espree] format 1`] = `
"Unexpected token \\u0066rom (1:15)
> 1 | import module \\u0066rom from "./module.wasm";
    |               ^
  2 |"
`;

exports[`valid-from-as-default-module-binding-escaped.mjs [flow] format 1`] = `
"Unexpected identifier, expected the keyword \`from\` (1:15)
> 1 | import module \\u0066rom from "./module.wasm";
    |               ^^^^^^^^^
  2 |"
`;

exports[`valid-from-as-default-module-binding-escaped.mjs [meriyah] format 1`] = `
"Import source must be string (1:17)
> 1 | import module \\u0066rom from "./module.wasm";
    |                 ^
  2 |"
`;

exports[`valid-from-as-default-module-binding-escaped.mjs [typescript] format 1`] = `
"Keywords cannot contain escape characters. (1:15)
> 1 | import module \\u0066rom from "./module.wasm";
    |               ^
  2 |"
`;

exports[`valid-from-as-default-module-binding-escaped.mjs format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
import module \\u0066rom from "./module.wasm";

=====================================output=====================================
import module from from "./module.wasm";

================================================================================
`;

exports[`valid-module-as-default-binding.mjs format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
import module from "./module.wasm";

=====================================output=====================================
import module from "./module.wasm";

================================================================================
`;

exports[`valid-module-as-default-binding-2.mjs format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
import module, { createRequire } from "node:module";

=====================================output=====================================
import module, { createRequire } from "node:module";

================================================================================
`;
